spec/filesystem.spec.js

```diff
@@ -446,6 +923,66 @@ describe('filesystem.spec: filesystem namespace tests', () => {
             assert.ok(typeof stats.createdAt == 'number');
             assert.ok(typeof stats.modifiedAt == 'number');
         });
+
+        it('throws an error for non-existent paths', async () => {
+            runner.run(`
+                try {
+                    let stats = await Neutralino.filesystem.getStats(NL_PATH + '/invalid-path');
+                }
+                catch(error) {
+                    await __close(error.code);
+                }
+            `);
+            assert.equal(runner.getOutput(), 'NE_FS_NOPATHE');
+        });
+
+        it('returns updated modifiedAt after file modification', async () => {
+            runner.run(`
+                await Neutralino.filesystem.writeFile(NL_PATH + '/.tmp/test-file.txt', 'Hello World');
+                let statsBefore = await Neutralino.filesystem.getStats(NL_PATH + '/.tmp/test-file.txt');
+                await new Promise(resolve => setTimeout(resolve, 1000));  // Ensure the timestamp changes
+                await Neutralino.filesystem.appendFile(NL_PATH + '/.tmp/test-file.txt', '!');
+                let statsAfter = await Neutralino.filesystem.getStats(NL_PATH + '/.tmp/test-file.txt');
+                await __close(JSON.stringify({ before: statsBefore, after: statsAfter }));
+            `);
+            let { before, after } = JSON.parse(runner.getOutput());
+            assert.ok(after.modifiedAt > before.modifiedAt);
+        });
+
+        it('returns stats for the file system root', async () => {
+            runner.run(`
+                let stats = await Neutralino.filesystem.getStats('/');
+                await __close(JSON.stringify(stats));
+            `);
+            let stats = JSON.parse(runner.getOutput());
+            assert.ok(typeof stats == 'object');
+            assert.ok(stats.isDirectory);
+            assert.ok(!stats.isFile);
```

Comment:
Could you please do a quick research and verify whether this works properly on Windows (on GitHub Actions also)? Thanks :tada:

Reply:
Hi @Sadaf-A, I removed this test case for now. I don't have a Windows machine to experiment with `/`'s behavior on Windows. Please check -- If `/` is a valid path and if it works properly on Windows, you can re-add the following code with another PR:

```js
        it('returns stats for the file system root', async () => {
            runner.run(`
                let stats = await Neutralino.filesystem.getStats('/');
                await __close(JSON.stringify(stats));
            `);
            let stats = JSON.parse(runner.getOutput());
            assert.ok(typeof stats == 'object');
            assert.ok(stats.isDirectory);
            assert.ok(!stats.isFile);
        });
```

I am not sure whether `/` refers to **root** on standard Windows systems.

Thanks :tada: